### PR TITLE
Migrate for Bazel's --incompatible_require_ctx_in_configure_features

### PR DESCRIPTION
**Pull request by hlopko**
This PR migrates rules_rust for https://github.com/bazelbuild/bazel/issues/7793

### CODE REVIEW

**Comment by hlopko** [comment]
It also means that rules_rust will require Bazel 0.25 at least.

File: rust/private/rustc.bzl [hunk 1]
```diff
@@ -140,11 +140,20 @@ def _get_linker_and_args(ctx, rpaths):
...
+            cc_toolchain = cc_toolchain,
+            requested_features = ctx.features,
+            unsupported_features = ctx.disabled_features,
+        )
+    else:
+        feature_configuration = cc_common.configure_features(
+            cc_toolchain = cc_toolchain,
+            requested_features = ctx.features,
+            unsupported_features = ctx.disabled_features,
+        )
```
Comment (damienmg) [inline rust/private/rustc.bzl]: Suggestion:

```python
    kwargs = {"ctx": ctx} if (len(BAZEL_VERSION) == 0 or versions.is_at_least("0.25.0", BAZEL_VERSION)) else None
    feature_configuration = cc_common.configure_features(
￼         cc_toolchain = cc_toolchain,
         requested_features = ctx.features,
         unsupported_features = ctx.disabled_features,
         **kwargs
￼     )
```

Comment (damienmg) [reply rust/private/rustc.bzl]: (remove parenthesis in the if, sorry for the mistake)

**Comment by hlopko** [comment]
Done, ptal :)

**Comment by hlopko** [comment]
Thanks for the quick review!

**Pull request merged**